cmd/otelcol/main_test.go

```diff
@@ -72,44 +74,44 @@ func TestRunFromCmdLine(t *testing.T) {
 		"NO_WINDOWS_SERVICE":      "true", // Avoid using the Windows service manager
 	}
 	for key, value := range requiredEnvVars {
-		os.Setenv(key, value)
-		defer os.Unsetenv(key)
+		t.Setenv(key, value)
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if tt.skipMsg != "" {
-				t.Skip(tt.skipMsg)
-			}
-
-			// GH darwin runners don't have docker installed, skip discovery tests on them
-			// given that the docker_observer is enabled by default.
-			if runtime.GOOS == "darwin" && (tt.name == "default_discovery" || tt.name == "dry-run_discovery") {
-				if os.Getenv("GITHUB_ACTIONS") == "true" {
-					t.Skip("skipping discovery tests on darwin runners since they don't have docker installed")
-				}
-			}
+			tt.run(t)
+		})
+	}
+}
 
-			testCtx, cancel := context.WithTimeout(context.Background(), tt.timeout)
-			defer cancel()
+func (tt runFromCmdLineTestCase) run(t *testing.T) {
+	if tt.skipMsg != "" {
+		t.Skip(tt.skipMsg)
+	}
 
-			otelcolCmdTestCtx = testCtx
-			defer func() {
-				otelcolCmdTestCtx = nil
-			}()
+	// GH darwin runners don't have docker installed, skip discovery tests on them
+	// given that the docker_observer is enabled by default.
+	if runtime.GOOS == "darwin" && (tt.name == "default_discovery" || tt.name == "dry-run_discovery") {
+		if os.Getenv("GITHUB_ACTIONS") == "true" {
+			t.Skip("skipping discovery tests on darwin runners since they don't have docker installed")
+		}
+	}
 
-			// Wait for the ConfigServer to be down after the test.
-			defer waitForPort(t, "55554")
```

Comment:
fyi - these were added for some strange flakiness in a test that I don't recall...